### PR TITLE
MB-4425 Filter Move Orders by Office User GBLOC in TOO

### PR DESCRIPTION
**Pull request by noelledusahel**
## Description

An TOO should only see move orders that have their same GBLOC, so apply a GBLOC filter to the move orders queue

## Reviewer Notes

Screen Shot 2020-10-07 at 3 26 46 PM
Screen Shot 2020-10-07 at 3 29 05 PM

## Setup
1. Open admin user app and create a new admin user, stop at the point that the form expands to include roles
```
make clean db_dev_e2e_populate
make server_run
make admin_client_run
```

2.  Bring up client user app and create a move
Choose a Transportation Office that is different from the one your chose for the  admin user. 
```
make client_run
```
2. On admin app, choose Tansportation Ordering Officer

4. Open Office App, sign in as TOO with your admin user
```
make office_client_run
```

5. The Move created should not appear on the list. 
## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](tbd) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._

### CODE REVIEW

File: pkg/services/move_order/move_order_fetcher.go [hunk 1]
```diff
@@ -14,17 +14,28 @@ type moveOrderFetcher struct {
...
 }
 
-func (f moveOrderFetcher) ListMoveOrders() ([]models.Order, error) {
+func (f moveOrderFetcher) ListMoveOrders(officeUserID uuid.UUID) ([]models.Order, error) {
 	// Now that we've joined orders and move_orders, we only want to return orders that
 	// have an associated move.
 	var moveOrders []models.Order
+	var transportationOffice models.TransportationOffice
+	// select the GBLOC associated with the transportation office of the session's current office user
+	f.db.Q().
```
Comment (duncan-truss) [inline pkg/services/move_order/move_order_fetcher.go:23]: I fixed this in my PR but I think there is an error being returned here we should check for

File: pkg/testdatagen/make_user.go [hunk 2]
```diff
@@ -9,10 +9,17 @@ import (
...
 	user := models.User{
-		LoginGovUUID:  uuid.Must(uuid.NewV4()),
-		LoginGovEmail: "first.last@login.gov.test",
-		Active:        true,
+		LoginGovUUID:           uuid.Must(uuid.NewV4()),
+		LoginGovEmail:          "first.last@login.gov.test",
+		Active:                 true,
+		CurrentMilSessionID:    milSessionID,
+		CurrentAdminSessionID:  adminSessionID,
+		CurrentOfficeSessionID: officeSessionID,
```
Comment (noelledusahel) [inline pkg/testdatagen/make_user.go:22]: forgot to revisit this, make this change due to a test error on the lack of reference to CurrentAdminSessionID on user

**Review by duncan-truss — APPROVED** [review]
Approved! Thanks for testing another TOO user with a different GBLOC.  Maybe in @monfresh 's PR we can add a db index on that `gbloc` column.

**Review by Ryan-Koch — APPROVED** [review]

**Comment by monfresh** [comment]
A couple of notes:

- Do we need to authorize the get orders handler endpoint? In my PR, I am only allowing office users with the TIO role to fetch the list of payment requests.
- I noticed that we never wrote a test for this endpoint. The `pkg/handlers/ghcapi/move_order_test.go` file does not test the `ListMoveOrdersHandler` function.

I will be pushing my PR up soon, which makes it easy to set the roles in the session for testing authorization.

File: pkg/services/move_order/move_order_fetcher_test.go [hunk 3]
```diff
@@ -71,27 +75,62 @@ func (suite *MoveOrderServiceSuite) TestListMoveOrders() {
 			Status: models.MTOShipmentStatusSubmitted,
 		},
 	})
+	officeUser := testdatagen.MakeOfficeUser(suite.DB(), testdatagen.Assertions{})
```
Comment (monfresh) [inline pkg/services/move_order/move_order_fetcher_test.go:78]: If we're not passing in any assertions, it's preferred to use the "default" maker, as in `testdatagen.MakeDefaultOfficeUser(suite.DB())`

File: pkg/services/move_order/move_order_fetcher_test.go [hunk 4]
```diff
@@ -71,27 +75,62 @@ func (suite *MoveOrderServiceSuite) TestListMoveOrders() {
...
+			TransportationOffice: models.TransportationOffice{
+				Gbloc: "AGFM",
+			},
+		})
+
+		order := testdatagen.MakeOrder(suite.DB(), testdatagen.Assertions{
+			OriginDutyStation: originDutyStation,
+		})
+
+		secondMoveTaskOrder := testdatagen.MakeMove(suite.DB(), testdatagen.Assertions{
```
Comment (monfresh) [inline pkg/services/move_order/move_order_fetcher_test.go:117]: Can we name this `secondMove` since it refers to the `Move` model?

File: pkg/services/move_order/move_order_fetcher_test.go [hunk 5]
```diff
@@ -71,27 +75,62 @@ func (suite *MoveOrderServiceSuite) TestListMoveOrders() {
...
+		testdatagen.MakeMTOShipment(suite.DB(), testdatagen.Assertions{
+			Move: secondMoveTaskOrder,
+			MTOShipment: models.MTOShipment{
+				Status: models.MTOShipmentStatusSubmitted,
+			},
+		})
+
+		moveOrders, err := moveOrderFetcher.ListMoveOrders(officeUser.ID)
+		moveOrder := moveOrders[0]
+		fmt.Println("GBLOC: ", moveOrder.OriginDutyStation.TransportationOffice)
```
Comment (monfresh) [inline pkg/services/move_order/move_order_fetcher_test.go:130]: Did you mean to leave this debugging statement in here?

Comment (noelledusahel) [reply pkg/services/move_order/move_order_fetcher_test.go:130]: ughhhhh my botched rebase comes back to haunt me

File: pkg/services/move_order/move_order_fetcher_test.go (hunk 3 above)
Comment (noelledusahel) [reply pkg/services/move_order/move_order_fetcher_test.go:78]: 👍

File: pkg/services/move_order/move_order_fetcher_test.go [hunk 6]
```diff
@@ -71,27 +75,62 @@ func (suite *MoveOrderServiceSuite) TestListMoveOrders() {
...
+		suite.FatalNoError(err)
+		suite.Len(moveOrders, 1)
```
Comment (monfresh) [inline pkg/services/move_order/move_order_fetcher_test.go:132]: Could we please change this to `suite.Equal(1, len(moveOrders))`? It's easier to debug. With `suite.Len`, it prints out the entire array, and you have to scroll a long time to the right to see what the actual length was. At least that's how it is in Goland.

File: pkg/testdatagen/make_office_user.go [hunk 7]
```diff
@@ -21,6 +21,10 @@ func MakeOfficeUser(db *pop.Connection, assertions Assertions) models.OfficeUser
 		user = MakeUser(db, assertions)
 	}
 
+	if assertions.User.CurrentAdminSessionID == "" {
+		assertions.User.CurrentAdminSessionID = "admin-session"
+	}
+
```
Comment (monfresh) [inline pkg/testdatagen/make_office_user.go:27]: What does this do? Why is this needed?

**Review by monfresh — COMMENTED** [review]
Looks good overall! I left some comments.

File: pkg/testdatagen/make_office_user.go (hunk 7 above)
Comment (noelledusahel) [reply pkg/testdatagen/make_office_user.go:27]: I was receiving an error in test (for `moveOrderFetcherTest#ListMoveOrders`) that the user was missing a relation to currentAdminSessionID, which i think may be tied to the session check in the handler? i made the assumption that an officeUser would have a admin session, but if this assumption is wrong i can probably pass this in as an assertion.
